### PR TITLE
MM-34508 in-app browser emm configuration

### PR DESCRIPTION
**Pull request by anurag6713**
#### Summary
Added support for EMM configuration for SSO flow switching between External Browser Flow and In-App WebView flow

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34508

#### Device Information
This PR was tested on: Android 11

### CODE REVIEW

**Review by enahum — APPROVED** [review]
Lgtm

**Comment by enahum** [comment]
Also cherry-pick for 1.41 and then submit a PR to bump version to 1.41.1 and build number. If you can't I'll do it when I'm back from my doc's appointment

**Review by enahum — APPROVED** [review]

**Review by enahum — APPROVED** [review]

**Review by adamjclarkson — APPROVED** [review]

**Review by furqanmlk — COMMENTED** [review]
**Personal:**
I am able to see Auth in browser 

image

### **Work:**
### Unable to open see Login page until `InAppSessionAuth` settings is `True`
image


image

**Comment by enahum** [comment]
@furqanmlk can you clarify, I did not get your comment

**Comment by furqanmlk** [comment]
> @furqanmlk can you clarify, I did not get your comment

Hi @enahum , I updated the screenshots.
In `Work` Environment, I am able to see Login screen only when `InAppSessionAuth` settings is `true`
if this setting is `false`, I am getting error

**Comment by enahum** [comment]
@anurag6713 looks like your code has some issues ;)

**Comment by anurag6713** [comment]
I think I understood the problem here. Work space doesn't have a browser app and I logged the error and it said "No Activity Found To Handle Intent".

Screenshot 2021-04-06 at 23 29 56
Screenshot 2021-04-06 at 23 30 13

May be having a browser in work tab can fix the issue... Will try to find how to get browser in that tab and test it.

**Comment by enahum** [comment]
@anurag6713 great find, but I would argue that because it can be possible we should handle this error accordingly and show something to the user so they can address it

**Comment by anurag6713** [comment]
@furqanmlk, I have tested in local and it's working with the following steps .. please follow them and confirm If external browser flow is working when inAppSessionAuth is set to `false`

Screenshot_1617780095

# Select chrome from the list: (you won't see it here because I have already added it)
Screenshot_1617780190

# Then you should see chrome in your work profile:
Screenshot_1617780197

**Comment by furqanmlk** [comment]
Thanks @anurag6713 
Yes after adding Chrome in Work space, I am able to see login in page on external browser with `inAppSessionAuth` to `false`
and with `true` it is opening in app session.

image

**Review by furqanmlk — APPROVED** [review]

File: app/screens/sso/sso_with_redirect_url.tsx [hunk 1]
```diff
@@ -57,12 +57,23 @@ function SSOWithRedirectURL({
...
 
-        const onError = () => setError(
-            intl.formatMessage({
-                id: 'mobile.oauth.failed_to_open_link',
-                defaultMessage: 'The link failed to open. Please try again.',
-            }),
-        );
+        const onError = (e: Error) => {
+            let message;
+            if (e?.message?.match(/no activity found to handle intent/i)) {
```
Comment (enahum) [inline app/screens/sso/sso_with_redirect_url.tsx:62]: This applies to Android only, right? What if the device is not in english? Does the message match?

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/screens/sso/sso_with_redirect_url.tsx (hunk 1 above)
Comment (anurag6713) [reply app/screens/sso/sso_with_redirect_url.tsx]: Yes the message is system generated and is matching different language ones too  ... Just now checked with Spanish.

**Review by enahum — APPROVED** [review]

**Pull request merged**